pandas/tests/io/test_common.py

```diff
@@ -411,3 +414,316 @@ def test_is_fsspec_url():
+        response_df = self.start_processing_headers()
+        self.send_header("Content-Type", "application/octet-stream")
+        self.end_headers()
+
+        # the fastparquet engine doesn't like to write to a buffer
+        # it can do it via the open_with function being set appropriately
+        # however it automatically calls the close method and wipes the buffer
+        # so just overwrite that attribute on this instance to not do that
+        def dummy_close():
+            pass
+
+        bio = BytesIO()
+        bio.close = dummy_close
+        # windows py38 np18 doesn't have 'snappy' installed
+        response_df.to_parquet(
+            "none",
+            index=False,
+            engine="fastparquet",
+            compression=None,
+            open_with=lambda x, y: bio,
+        )
+        response_bytes = bio.getvalue()
+
+        self.write_back_bytes(response_bytes)
+
+
+class PickleUserAgentResponder(BaseUserAgentResponder):
+    def do_GET(self):
+        response_df = self.start_processing_headers()
+        self.send_header("Content-Type", "application/octet-stream")
+        self.end_headers()
+
+        bio = BytesIO()
+        response_df.to_pickle(bio)
+        response_bytes = bio.getvalue()
+
+        self.write_back_bytes(response_bytes)
+
+
+class StataUserAgentResponder(BaseUserAgentResponder):
+    def do_GET(self):
+        response_df = self.start_processing_headers()
+        self.send_header("Content-Type", "application/octet-stream")
+        self.end_headers()
+
+        bio = BytesIO()
+        response_df.to_stata(bio, write_index=False)
+        response_bytes = bio.getvalue()
+
+        self.write_back_bytes(response_bytes)
+
+
+class AllHeaderCSVResponder(http.server.BaseHTTPRequestHandler):
+    """
+    Send all request headers back for checking round trip
+    """
+
+    def do_GET(self):
+        response_df = pd.DataFrame(self.headers.items())
+        self.send_response(200)
+        self.send_header("Content-Type", "text/csv")
+        self.end_headers()
+        response_bytes = response_df.to_csv(index=False).encode("utf-8")
+        self.wfile.write(response_bytes)
+
+
+@pytest.mark.parametrize(
+    "responder, read_method, port, parquet_engine",
+    [
+        (CSVUserAgentResponder, pd.read_csv, 34259, None),
+        (JSONUserAgentResponder, pd.read_json, 34260, None),
+        (ParquetPyArrowUserAgentResponder, pd.read_parquet, 34268, "pyarrow"),
+        (ParquetFastParquetUserAgentResponder, pd.read_parquet, 34273, "fastparquet"),
+        (PickleUserAgentResponder, pd.read_pickle, 34271, None),
+        (StataUserAgentResponder, pd.read_stata, 34272, None),
+        (GzippedCSVUserAgentResponder, pd.read_csv, 34261, None),
+        (GzippedJSONUserAgentResponder, pd.read_json, 34262, None),
+    ],
+)
+def test_server_and_default_headers(responder, read_method, port, parquet_engine):
+    if parquet_engine is not None:
+        if parquet_engine == "pyarrow":
```

Comment:
Would `pytest.importorskip` work here instead of try-except?

Reply:
I had that in a prior commit and it seemed like it still ran the test, so I replaced it with a try-except.  It must have been confounded with something else I hadn't quite worked out.  I'll give that a go again with ``pytest.importorskip``, sorry about that!